tests/test_diffraction_objects.py

```diff
@@ -402,114 +403,107 @@ def test_dump(tmp_path, mocker):
+                        [2.0, 1.0 / np.sqrt(2), 90.0, np.sqrt(2) * 2 * np.pi],
+                        [3.0, 1.0, 180.0, 1.0 * 2 * np.pi],
+                    ]
+                ),
+                "metadata": {},
+                "_input_xtype": "tth",
+                "name": "",
+                "scat_quantity": "",
+                "qmin": np.float64(0.0),
+                "qmax": np.float64(1.0),
+                "tthmin": np.float64(0.0),
+                "tthmax": np.float64(180.0),
+                "dmin": np.float64(2 * np.pi),
+                "dmax": np.float64(np.inf),
+                "wavelength": 4.0 * np.pi,
+            },
+            True,
+        ),
+        (  # instantiate just array attributes
+            {
+                "xarray": np.array([np.inf, 2 * np.sqrt(2) * np.pi, 2 * np.pi]),
+                "yarray": np.array([1.0, 2.0, 3.0]),
+                "xtype": "d",
+                "wavelength": 4.0 * np.pi,
+                "scat_quantity": "x-ray",
+            },
+            {
+                "_all_arrays": np.array(
+                    [
+                        [1.0, 0.0, 0.0, np.float64(np.inf)],
+                        [2.0, 1.0 / np.sqrt(2), 90.0, np.sqrt(2) * 2 * np.pi],
+                        [3.0, 1.0, 180.0, 1.0 * 2 * np.pi],
+                    ]
+                ),
+                "metadata": {},
+                "_input_xtype": "d",
+                "name": "",
+                "scat_quantity": "x-ray",
+                "qmin": np.float64(0.0),
+                "qmax": np.float64(1.0),
+                "tthmin": np.float64(0.0),
+                "tthmax": np.float64(180.0),
+                "dmin": np.float64(2 * np.pi),
+                "dmax": np.float64(np.inf),
+                "wavelength": 4.0 * np.pi,
+            },
+            False,
+        ),
+    ],
 )
-def test_init_valid(init_args, expected_do_dict):
-    actual_do_dict = DiffractionObject(**init_args).__dict__
+def test_init_valid(do_init_args, expected_do_dict, divide_by_zero_warning_expected):
+    if divide_by_zero_warning_expected:
+        with pytest.warns(RuntimeWarning, match="divide by zero encountered in divide"):
+            actual_do_dict = DiffractionObject(**do_init_args).__dict__
+    else:
+        actual_do_dict = DiffractionObject(**do_init_args).__dict__
     diff = DeepDiff(
         actual_do_dict, expected_do_dict, ignore_order=True, significant_digits=13, exclude_paths="root['_id']"
     )
     assert diff == {}
 
 
-test_init_invalid_params = [
-    (  # UC1: no arguments provided
-        {},
-        "missing 3 required positional arguments: 'xarray', 'yarray', and 'xtype'",
-    ),
-    (  # UC2: only xarray and yarray provided
-        {"xarray": np.array([0.0, 90.0]), "yarray": np.array([0.0, 90.0])},
-        "missing 1 required positional argument: 'xtype'",
-    ),
-]
-
-
-@pytest.mark.parametrize("init_args, expected_error_msg", test_init_invalid_params)
+@pytest.mark.parametrize(
```

Comment:
I will further refine these comments with higher level commets, etc. in the following PRs (issues have been created for these)